python/lsst/ts/eas/eas_csc.py

```diff
@@ -84,290 +88,238 @@ def __init__(
+            }
+        )
 
-        self.old_valve_position: float = float("nan")
+    async def handle_summary_state(self) -> None:
+        """Override of the handle_summary_state function to
+        set up the control loop.
+        """
+        self.log.debug(f"handle_summary_state {salobj.State(self.summary_state).name}")
 
-        self.log.info("__init__")
+        if self.disabled_or_enabled:
+            if self.health_monitor_task.done():
+                self.health_monitor_task = asyncio.create_task(self.health_monitor())
 
-    async def connect(self) -> None:
-        """Connect the EAS CSC or start the mock client, if in
-        simulation mode.
-        """
-        self.log.info("Connecting")
-        self.log.info(self.config)
-        self.log.info(f"self.simulation_mode = {self.simulation_mode}")
-        if self.config is None:
-            raise RuntimeError("Not yet configured")
-        if self.connected:
-            raise RuntimeError("Already connected")
-        if self.simulation_mode == 1:
-            # TODO Add code for simulation case, see DM-26440
-            pass
         else:
-            # TODO Add code for non-simulation case
+            await self.health_monitor_shutdown()
+
+    async def health_monitor_shutdown(self) -> None:
+        self.health_monitor_task.cancel()
+        try:
+            await self.health_monitor_task
+        except asyncio.CancelledError:
             pass
 
-        if self.eas:
-            self.eas.connect()
+    async def close_tasks(self) -> None:
+        """Stop active tasks."""
+        await self.health_monitor_shutdown()
+        await super().close_tasks()
 
-    async def disconnect(self) -> None:
-        """Disconnect the EAS CSC, if connected."""
-        self.log.info("Disconnecting")
+    async def configure(self, config: SimpleNamespace) -> None:
+        self.config = config
 
-        if self.eas:
-            self.eas.disconnect()
+    async def health_monitor(self) -> None:
+        """Manages the `monitor_dome_shutter` control loop.
 
-    async def handle_summary_state(self) -> None:
-        """Override of the handle_summary_state function to connect or
-        disconnect to the EAS CSC (or the mock client) when needed.
+        Manages the `monitor_dome_shutter` control loop with backoff and
+        time-based failure reset, which is (hopefully) robust against
+        disconnects of the remotes, as well as any other form of
+        recoverable failure.
         """
-        self.log.info(f"handle_summary_state {salobj.State(self.summary_state).name}")
+        self.log.debug("health_monitor")
+        failure_count = 0
+        last_attempt_time = utils.current_tai()
 
-        if self.disabled_or_enabled:
-            # Start the thermal task if not already started
-            if self.m1m3_thermal_task.done() and self.simulation_mode == 0:
-                self.m1m3_thermal_task = asyncio.create_task(self.run_control())
+        while True:
```

Comment:
Beware of using `while True` this can lead to dangling background method that never terminate. Consider adding a termination condition here like `while self.disable_or_enabled`, or something like that.